DaiMika/MXVirtualFit
# Add detailed comments to all code files

This pull request adds a new feature for user authentication, allowing users to register, log in, and log out. This feature includes the following changes:

1. User Registration:
   - Added a new registration form.
   - Integrated with the backend to store user credentials securely in the database.
   
2. User Login:
   - Added a login form.
   - Implemented session management to keep users logged in.
   
3. User Logout:
   - Added a logout functionality that clears the user session.

Changes
- frontend/src/components/Register.js**: Created a registration form component.
- frontend/src/components/Login.js**: Created a login form component.
- frontend/src/components/Logout.js**: Created a logout button component.
- backend/routes/auth.js**: Added routes for user registration, login, and logout.
- backend/models/User.js**: Defined the User model schema.

How to Test
1. Pull the branch and run the backend server.
   ```sh
   cd backend
   python app.py

2.  Run the frontend development server.
cd frontend
npm start

3.  Navigate to the registration page and create a new account.

4.  Log in with the newly created account.

5.  Verify that the user is redirected to the home page after login.

6.  Click the logout button and ensure the user session is cleared.

Related Issues
Resolves #123: User Authentication
Resolves #124: User Session Management

NOTE: Ensure to run database migrations before testing:  flask db upgrade
           Make sure to set the environment variables in the .env file.
